### DISCUSSION TITLE
Can't add Excel and PDF button in my Serenity Form

### DISCUSSION THREAD

**Discussion by AHSOFFICE** [serenity-is/Serenity#7169] [Help] [answered]
When i try to add button in Grid.ts file , First  Jquery constructor automatically not there and when i manually add it says serenity.extension.toolbar not there.

**Comment by VictorTomaili** [comment]
Did you update your npm packages?

The latest serenity doesn't have jquery dependency.

**Reply by AHSOFFICE** [reply]
![Captures](https://github.com/serenity-is/Serenity/assets/129037734/1c468ff8-ab7c-465a-9d83-2f2209eabd13)
Don't know yet, coz newbie in this framework, can you please help me by see image.

**Reply by VictorTomaili** [reply]
open commandline, go to the directory `~/MyProject/MyProject.Web` run the `npm i` command

**Comment by AHSOFFICE** [comment]
![Captures](https://github.com/serenity-is/Serenity/assets/129037734/d01ca8fb-d9b9-4ff1-b3d2-91f40e6ef765)

Run successfully! now please let me know further process?

**Comment by AHSOFFICE** [comment]
![Capture](https://github.com/serenity-is/Serenity/assets/129037734/1e86d481-f17f-4996-a0a8-b75f8c6c96b8)
 Jquery constuctor not generated when i make module with sergen , nor add any button

**Reply by VictorTomaili** [reply]
it is not required for the basic grids

**Reply by AHSOFFICE** [reply]
![Capture](https://github.com/serenity-is/Serenity/assets/129037734/fa9bd8ba-51c6-4f5a-a0d5-958148deff67)
 I,m using serene framword still this says serene cannot find.

**Reply by VictorTomaili** [reply, marked as answer]
https://github.com/serenity-is/common-features/blob/237433ccb7938628d0a3c9731c3b843cd914d3c3/src/Serenity.Demo.Northwind/Modules/Order/OrderGrid.ts#L47

check the samples

**Reply by AHSOFFICE** [reply]
Thank you VictorTomaili........ Working button.....

Last query , can we generate reports in serenity like we generate module with sergen.

**Reply by VictorTomaili** [reply]
no, you need to write report manually.

**Reply by AHSOFFICE** [reply]
Can you share any sample , like earlier you send me in which we get to know how to make reports in it

**Reply by AHSOFFICE** [reply]
Can we do Order-order detail in one form in serenity free version?[VictorTomaili](https://github.com/VictorTomaili)

**Reply by VictorTomaili** [reply]
You can find many samples here https://github.com/serenity-is/common-features

**Reply by AHSOFFICE** [reply]
**When i add editor on Form file** [VictorTomaili](https://github.com/VictorTomaili)
![form](https://github.com/serenity-is/Serenity/assets/129037734/9b0fddc2-cc1f-4ee9-a74d-dcaf171ffa48)

**and make editor in Import>>ClientType folder**
![editor](https://github.com/serenity-is/Serenity/assets/129037734/e50bc9fc-d400-4d95-ac67-3acd432e88b6)


**and when i run and click on form it gives this error**
![Capture](https://github.com/serenity-is/Serenity/assets/129037734/134bf7ff-22ff-4fea-9b3d-d469e2f913c8)
